### PR TITLE
python3Packages.aiorun: init at 2019.11.1

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

Full test suite runs.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by costrouc** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

**Comment by costrouc** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

**Review by jonringer — APPROVED** [review]
diff LGTM
commit LGTM
has tests :+1: 

```
[4 built, 12 copied (18.9 MiB), 2.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/75797
2 package were built:
python37Packages.aiorun python38Packages.aiorun
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

File: pkgs/development/python-modules/aiorun/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  checkInputs = [
+    pytest
+    pytestcov
+    uvloop
+  ];
+
+  checkPhase = ''
+    pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/aiorun/default.nix]:
```suggestion
    HOME=$TMPDIR pytest
```
```
PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
builder for '/nix/store/lmnxci3xvpp4qkyk0gihcyvvlv1pz4sr-python3.8-aiorun-2019.11.1.drv' failed with exit code 1
error: build of '/nix/store/frd56ynplxvr5c2m728kkl8z2qpasjjd-python3.7-aiorun-2019.11.1.drv', '/nix/store/lmnxci3xvpp4qkyk0gihcyvvlv1pz4sr-python3.8-aiorun-2019.11.1.drv' failed
```

Comment (costrouc) [reply pkgs/development/python-modules/aiorun/default.nix]: So we no longer need to `HOME=$(mktemp -d)`? This is nice!

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

File: pkgs/development/python-modules/aiorun/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "cjrh";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "04p3sci6af6qqfkcqamsqhmqqrigcwvl4bmx8yv5ppvkyq39pxi7";
+  };
+
```
Comment (jonringer) [inline pkgs/development/python-modules/aiorun/default.nix]: sorry, the `/homeless-shelter` error was actually during flit build step
```suggestion
    # allow for writable directory for darwin
  preBuild = ''
    export HOME=$TMPDIR
  '';
```

File: pkgs/development/python-modules/aiorun/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  checkInputs = [
+    pytest
+    pytestcov
+    uvloop
+  ];
+
+  checkPhase = ''
+    HOME=$TMPDIR pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/aiorun/default.nix]:
```suggestion
    pytest
```

File: pkgs/development/python-modules/aiorun/default.nix (hunk 2 above)
Comment (costrouc) [reply pkgs/development/python-modules/aiorun/default.nix]: fixed.

File: pkgs/development/python-modules/aiorun/default.nix (hunk 3 above)
Comment (costrouc) [reply pkgs/development/python-modules/aiorun/default.nix]: fixed.

**Comment by costrouc** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.aiorun python38Packages.aiorun

**Comment by jonringer** [comment]
builds on darwin

**Pull request merged**